File: multiply-strings/multiply-strings.py
class Solution:
    def multiply(self, num1: str, num2: str) -> str:
        sum1 = 0
        sum2 = 0
        for c in num1: #Convert num1 to int
            sum1 *= 10
            sum1 += int(c)
        for c in num2: #Convert num2 to int
            sum2 *= 10
            sum2 += int(c)
        return str(sum1 * sum2)

File: multiply-strings/multiply-strings.java
class Solution 
{
    public String multiply(String num1, String num2) 
    {
        // Multiplication by 0
        if (num1.equals("0") || num2.equals("0"))
            return "0";
        
        // Create an array to store products
        int[] answer = new int[num1.length() + num2.length() - 1];
        
        // Perform basic multiplication for every value
        for (int i = 0; i < num1.length(); i++)
            for (int j = 0; j < num2.length(); j++)
                answer[i+j] += (num1.charAt(i) - '0') * (num2.charAt(j) - '0');
        
        // Shift the products up the array using basic addition
        for (int i = answer.length-1; i > 0; i--)
        {
            answer[i-1] += answer[i] / 10;
            answer[i] %= 10;
        }
        
        // Convert the int array into a string result
        StringBuilder build = new StringBuilder();
        for (int num : answer)
            build.append(num);
        
        return build.toString();
    }
}